### PR TITLE
[SYCL] fix a bug in annotated property check

### PR DESCRIPTION
**Pull request by broxigarchen**
Fixed a bug in FPGA annotated property check such that the logic should be 'or' instead of 'and'

### CODE REVIEW

**Comment by dm-vodopyanov** [comment]
Can you please add a test?

**Comment by broxigarchen** [comment]
> Can you please add a test?

There is a test source using this. This PR also fix it

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by broxigarchen** [comment]
@intel/llvm-gatekeepers can you help to take a look at this? Thanks!

File: sycl/test/extensions/annotated_arg/annotated_arg_for_ptr.cpp
Comment (tiwaria1) [inline sycl/test/extensions/annotated_arg/annotated_arg_for_ptr.cpp:1]: Please add a line that checks that using alignment on annotated_ptr without buffer location does NOT error out.

**Review by tiwaria1 — CHANGES_REQUESTED** [review]
Requested a test change

**Review by tiwaria1 — APPROVED** [review]

**Comment by broxigarchen** [comment]
> @intel/llvm-gatekeepers can you help to take a look at this? Thanks!

Gentle ping again!

**Comment by steffenlarsen** [comment]
ESIMD Timed Out Tests (1):
  SYCL :: Prefix_Local_sum3.cpp - Unrelated. Expected to be a hiccup.

**Pull request merged**